### DISCUSSION TITLE
x-data not correctly registering on tbody classes

### DISCUSSION THREAD

**Discussion by HeartthrobRob** [alpinejs/alpine#3773] [1. Help]
Alpine.js behaviour requires x-data to be enabled on parent element.
When parent = tbody, this doesn't seem to always apply evenly.

Example:
```html
<tbody x-data="aFunctionHere()">
    <div x-data>
        <p x-text="$store.some_global_variable"></p>
    </div>
</tbody>
```

Without the `<div x-data>` line the x-text will not be displayed at all. 
I assume this is because of a conflict w the tbody class.

**Comment by ekwoka** [comment]
There is no code in your post.

**Reply by SimoTod** [reply]
updated, he didn't wrap the code within ```

**Comment by SimoTod** [comment]
you can't use a div or p tag as a direct child of a tbody. The browser is likely to move it out so when Alpine runs, it's not aware that the div and p were meant to inherit the tbody scope.

**Reply by ekwoka** [reply]
To add to the above:

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tbody

The only valid child is tr.

So @HeartthrobRob Try with proper HTML and see if that solves the issue?
